device_backends/Subdevice/src/SubdeviceRegisterAccessor.cc

```diff
@@ -26,9 +26,11 @@ namespace ChimeraTK {
 
   bool SubdeviceRegisterAccessor::doWriteTransfer(ChimeraTK::VersionNumber) {
     std::lock_guard<decltype(_backend->mutex)> lockGuard(_backend->mutex);
+    // This is "_numberOfWords / _accData->getNumberOfSamples()" rounded up:
+    auto nTransfers = (_numberOfWords + _accData->getNumberOfSamples() - 1) / _accData->getNumberOfSamples();
     try {
       size_t idx = 0;
-      for(size_t adr = _byteOffset; adr < _byteOffset + 4 * _numberOfWords; adr += 4) {
+      for(size_t adr = _startAddress; adr < _startAddress + nTransfers; ++adr) {
         if(_backend->type == SubdeviceBackend::Type::threeRegisters) {
```

Comment:
I know of at least two places with a 1-D register (four 32-bit words) where I have to wait for the busy register to clear after each word that is written. Previously only one word per address was written, so it does not make a difference where the code is waiting for the flag to clear.

I am not sure how the firmware is implemented in this case, and if it is the same in all cases, but there are two possibilities where the code might have to wait:
* after each word
* only after writing everything that belongs to one address (this implementation)

Maybe needs clarification with the FW team.

Reply:
That would be a different use case. In my understanding, the case you are mentioning does not have an address register at all, because the address unfolding is already done in firmware. That would hence be more like an extension to the area mode of the SubdeviceBackend (this feature is only about the 2regs/3regs mode). Needing to wait after each word of the data register in 2/3regs mode sounds very weird, I don't think that such a case exists (clearly not the use case currently at hand). If we need this in future, we can still extend by adding a new parameter.